README.markdown

```diff
@@ -1320,257 +1798,267 @@ $my_path3 = ['C:/Program Files (x86)/Puppet Labs/Puppet','C:/Program Files/Puppe
 
-  The following values will pass:
+The following values will pass:
 
-  ~~~
-  $my_hash = { 'one' => 'two' }
-  validate_hash($my_hash)
-  ~~~
+```puppet
+$my_hash = { 'one' => 'two' }
+validate_hash($my_hash)
+```
 
-  The following values will fail, causing compilation to abort:
+The following values will fail, causing compilation to terminate:
 
-  ~~~
-  validate_hash(true)
-  validate_hash('some_string')
-  $undefined = undef
-  validate_hash($undefined)
-  ~~~
+```puppet
+validate_hash(true)
+validate_hash('some_string')
+$undefined = `undef`
+validate_hash($undefined)
+```
 
 *Type*: statement.
 
 #### `validate_integer`
 
 **Deprecated. Will be removed in a future version of stdlib. See [`validate_legacy`](#validate_legacy).**
 
-Validates that the first argument is an integer (or an array of integers). Aborts catalog compilation if any of the checks fail.
-
-  The second argument is optional and passes a maximum. (All elements of) the first argument has to be less or equal to this max.
-
-  The third argument is optional and passes a minimum. (All elements of) the first argument has to be greater or equal to this min.
-  If, and only if, a minimum is given, the second argument may be an empty string or undef, which will be handled to just check
-  if (all elements of) the first argument are greater or equal to the given minimum.
-
-  It will fail if the first argument is not an integer or array of integers, and if arg 2 and arg 3 are not convertable to an integer.
-
-  The following values will pass:
-
-  ~~~
-  validate_integer(1)
-  validate_integer(1, 2)
-  validate_integer(1, 1)
-  validate_integer(1, 2, 0)
-  validate_integer(2, 2, 2)
-  validate_integer(2, '', 0)
-  validate_integer(2, undef, 0)
-  $foo = undef
-  validate_integer(2, $foo, 0)
-  validate_integer([1,2,3,4,5], 6)
-  validate_integer([1,2,3,4,5], 6, 0)
-  ~~~
-
-  * Plus all of the above, but any combination of values passed as strings ('1' or "1").
-  * Plus all of the above, but with (correct) combinations of negative integer values.
-
-  The following values will fail, causing compilation to abort:
-
-  ~~~
-  validate_integer(true)
-  validate_integer(false)
-  validate_integer(7.0)
-  validate_integer({ 1 => 2 })
-  $foo = undef
-  validate_integer($foo)
-  validate_integer($foobaridontexist)
-
-  validate_integer(1, 0)
-  validate_integer(1, true)
-  validate_integer(1, '')
-  validate_integer(1, undef)
-  validate_integer(1, , 0)
-  validate_integer(1, 2, 3)
-  validate_integer(1, 3, 2)
-  validate_integer(1, 3, true)
-  ~~~
-
-  * Plus all of the above, but any combination of values passed as strings ('false' or "false").
-  * Plus all of the above, but with incorrect combinations of negative integer values.
-  * Plus all of the above, but with non-integer items in arrays or maximum / minimum argument.
-
-  *Type*: statement.
+Validates an integer or an array of integers. Terminates catalog compilation if any of the checks fail.
```

Comment:
Can we just remove all documentation except the line stating that it is deprecated? We don't need to explain how to use deprecated functionality.